### DISCUSSION TITLE
应用市场安装JumpServer失败问题

### DISCUSSION THREAD

**Discussion by JAINKRE** [1Panel-dev/1Panel#5417] [Q&A]
### 1Panel 版本

v1.10.10-lts

### 问题描述

使用应用市场默认安装配置：
![screenshot-20240611-104740](https://github.com/1Panel-dev/1Panel/assets/63288007/231441fe-76d9-43b7-9ea0-d97a6fe1c126)
> 防火墙已放开9999端口

面板是安装状态显示`成功`，但是访问 http://192.168.3.88:9999/ 失败，日志如下：
![screenshot-20240611-104907](https://github.com/1Panel-dev/1Panel/assets/63288007/c7cac544-b487-45ef-b742-85aca5f0f7e9)

## BUG解决

安装时修改`compose 文件`中的 `DB_HOST: ${PANEL_DB_HOST}`和` REDIS_HOST: ${REDIS_HOST}`为主机地址，安装成功。

### 重现步骤

应用商店 > 安装`JumpServer`

**Comment by wanghe-fit2cloud** [comment]
可以先手动启动一个容器，然后通过使用 `ping` 或者 `nc` 等命令来验证 MySQL 和 Redis 服务名称是否可用。

**Reply by JAINKRE** [reply]
这个问题我已经解决了：
 
> 安装时修改compose 文件中的 DB_HOST: ${PANEL_DB_HOST}和 REDIS_HOST: ${REDIS_HOST}为主机地址，安装成功。

其实提这个issue本来是想作为一个bug的。应用商店默认安装时`PANEL_DB_HOST` 和 `REDIS_HOST` 地址获取不到，日志报错。手动修改compose文件这两个参数为本机IP就可以正常安装了。

所以，本意提这个issue是希望可以修复一下
> 应用商店默认安装时`PANEL_DB_HOST` 和 `REDIS_HOST` 地址获取不到

**Reply by wanghe-fit2cloud** [reply]
不是的，改成本机 IP 的前提是要应用对外暴露端口。
目前的两个 host 是应用的服务名称，我们在本地环境均未能复现上述连接失败的问题。
> 推测可能的原因是 IP 段冲突导致的容器网络不通，你可以重新生成一下 1panel-network 这个容器网络试一下。

**Reply by ha1o0** [reply]
> 不是的，改成本机 IP 的前提是要应用对外暴露端口。 目前的两个 host 是应用的服务名称，我们在本地环境均未能复现上述连接失败的问题。
> 
> > 推测可能的原因是 IP 段冲突导致的容器网络不通，你可以重新生成一下 1panel-network 这个容器网络试一下。

我也遇到这个问题了，包括 用 1panel 安装的 psql 也是这样， 应用（比如 halo）如果用到这个 psql，在 docker-compose 构建容器后， 里面用的 .env 中的 dbhost 环境变量无法连接到 psql，解决方法是 docker-compose 中改 dbhost 为本机 ip 且开放 psql 的外部端口访问后就可以连上了。
